contracts/src/systems/combat/contracts/battle_systems.cairo

```diff
@@ -658,6 +683,11 @@ mod battle_systems {
                 world.write_model(@army);
                 world.write_model(@army_health);
                 world.write_model(@army_quantity);
+            } else if army.troops.count() > 0 {
```

Comment:
Nice one needs a double check, the expectation is leaving a winning battle

Reply:
the `BattleLord` achievement should only  be inside the internal `fn leave_battle`. inside the function, check if the army left early then give the reward if the army didnt leave early and they are still alive. it should be at the bottom of the function. then you can remove the code from everywhere else